Made the following changes in github_traffic_stats.py

dapper-hacker opened the pull request:
-added import datetime to parse github response for datetime object
-parsed the datetime json object to a time-format string
-added the following condition in the collect method
db.get(timestamp) is False
to qualify if no object exists in pickleDB

Made the following changes in README.md
-Based on timestamp formatting, the README examples had to be updated to show the new response format so that users are not confused.

dapper-hacker committed:
Made the following changes in github_traffic_stats.py
-added import datetime to parse github response for datetime object
-parsed the datetime json object to a time-format string
-added the following condition in the collect method
db.get(timestamp) is False
to qualify if no object exists in pickleDB

Made the following changes in README.md
-Based on timestamp formatting, the README examples had to be updated to show the new response format so that users are not confused.

seladb commented on README.md:
```diff
@@ -82,25 +82,22 @@ python github_traffic_stats.py collect -r pcapplusplus -t ****** -o seladb
-2018-02-21T00:00:00Z {"count": 150, "uniques": 50}
-2018-02-22T00:00:00Z {"count": 150, "uniques": 35}
-18 elements
+2020-04-27 {"count": 7, "uniques": 6}
+2020-04-28 {"count": 119, "uniques": 59}
+2020-04-29 {"count": 143, "uniques": 58}
+2020-04-30 {"count": 186, "uniques": 55}
+2020-05-01 {"count": 77, "uniques": 40}
+2020-05-02 {"count": 49, "uniques": 33}
+2020-05-03 {"count": 61, "uniques": 22}
+2020-05-04 {"count": 120, "uniques": 63}
+2020-05-05 {"count": 172, "uniques": 73}
+2020-05-06 {"count": 110, "uniques": 47}
+2020-05-07 {"count": 108, "uniques": 49}
+2020-05-08 {"count": 110, "uniques": 41}
+2020-05-09 {"count": 76, "uniques": 26}
+2020-05-10 {"count": 86, "uniques": 34}
+2020-05-11 {"count": 59, "uniques": 37}
+15 elements
```
are these changes referring to your library? I believe PcapPlusPlus has different numbers...

dapper-hacker replied:
You are correct. These traffic results are from my target repository what I am monitoring. Not from PcapPlusPlus since I cannot pull the traffic metrics from it. 
My intent was to show updated response format so that the end user doesnt get confused while reading README vs Running the updated code. 
The README sample traffic response cant be reproduced anyways due to the moving 14 day window on github.

seladb replied:
please see my other comment regarding changing the date format

seladb commented on github_traffic_stats.py:
```diff
@@ -26,10 +27,13 @@ def collect(db, user, passwd, token, repo, org):
     for view_per_day in views_14_days['views']:
         timestamp = view_per_day['timestamp']
+        timestamp = datetime.datetime.strptime(timestamp,"%Y-%m-%dT%H:%M:%SZ")
+        timestamp = timestamp.strftime(time_format)
```
I just noticed you're also changing the date format. This will make the new version non backward compatible to the older version which might be an issue (at least for me 😃). So I see 2 options here:

- Revert the date format change
- Add a command line parameter to define the date format. The default should be the one in the current version

dapper-hacker replied:
Hi, 
So, I need a clarification. When you say revert the date format change, in this scenerio, 
1. Is it acceptable to convert the python unicode object ([timestamp] which caused the issue to begin with) into  python timestamp object AND keep the format same as you had?

seladb replied:
Yes, this is correct. Let's just make sure the date time format stays the same

dapper-hacker replied:
completed the request. You can review.
1. reverted back the changes in README. Its now same as your master README
2. Changed the timestamp variable from unicode object to string so that it could be used as a key in pickedb while keeping the output format as requested.

dapper-hacker committed:
- Reverting the changes in README
- Fixing the changes to timestamp variable to be of type string and format as requested.

seladb commented:
I just upgraded this project to python 3 and fixed this issue also. I'm closing the PR now.

Thanks for your contribution!
